### ISSUE TITLE
Supporting mobile

### ISSUE THREAD

**Issue by juho-hanhimaki** [SteveSandersonMS/WebWindow#51]
What about mobile (Android / iOS)? Could this technology be adapted to also support building mobile apps. It would be exciting if one could build existing Blazor app to mobile app. 

It would be a sort of alternative to PWA. The PWA experience is still not super great on most platforms. Hopefully it will get there so there would be no need for alternate solutions but oh well.

**Comment by x3haloed** [comment]
I think the biggest problem with PWAs is that they don’t get to take advantage of the discoverability or sense of security provided by the App Store or Play Store, and for that reason, I think this is a legitimate and enticing idea.

**Comment by scott-scott** [comment]
That would require support for Android and iOS from .net core, of which there currently is none.

[https://github.com/dotnet/core/blob/master/release-notes/3.1/3.1-supported-os.md](https://github.com/dotnet/core/blob/master/release-notes/3.1/3.1-supported-os.md)

**Comment by x3haloed** [comment]
Targeting .NET Standard and implementing on Xamarin or Mono won't work? Please pardon my ignorance; I haven't had a chance to look through the source code here.

**Comment by juho-hanhimaki** [comment]
> That would require support for Android and iOS from .net core, of which there currently is none.
> 
> https://github.com/dotnet/core/blob/master/release-notes/3.1/3.1-supported-os.md
> 
> edit: what about using [this](https://github.com/Daddoon/BlazorMobile)? unless I'm misunderstanding what you are after.

BlazorMobile is cool but it's not web based. Unfortunately it would require rewrite of web app into a mobile app (the UI portion) which can be a ton of work for large web app. Instead I want to reuse the web app code to build a mobile app. PWA would be the ideal solution if it had proper support and discoverability on all platforms. As it stands there are still scenarios where native app wins.

**Comment by x3haloed** [comment]
Looks like WebWindow already targets .NET Standard -- netstandard2.1, to be exact. In this case, it should definitely be possible to implement on mobile via Xamarin.Android v10.0 and Xamarin.iOS v12.16. I'm not sure how easy or hard it would be to get "native" builds working...

**Comment by aherrick** [comment]
@x3haloed Agreed. That's likely the approach I would take. Forms App with a full screen WebView.

**Comment by denisbredikhin** [comment]
Hello guys. Any plans from MS to work in this direction?
I've checked BlazorMobile, by recommendation of @scott-scott, but the difference is that it is using WASM runtime, and not a full .NET (Core or Xamarin). And for my task I need a full one (WCF client).

**Comment by vanderstack** [comment]
I have an application I'm working on right now which I had been building as a PWA so that I could have the 'feel' of an app while still working in Blazor, but ultimately my goal is to be able to publish to the app stores and windows store. It would be amazing if this prototype could be extended to allow the development of a fully cross-platform WebView based Blazor application.
